### PR TITLE
fixed all the bugs related to firmware repository

### PR DESCRIPTION
**Pull request by AnikaAgiwal2711**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 80% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Backward compatibility is not broken

# How Has This Been Tested?
This is a bug fix PR . So I ran the specific tests manually for which the bugs were raised. Also few functional testcases were also run

### CODE REVIEW

**Comment by shenda1** [comment]
@AnikaAgiwal2711  Can you update how its been tested?

File: powerflex/helper/firmware_repository_helper.go [hunk 1]
```diff
@@ -33,6 +33,11 @@ func UpdateFrimwareRepositoryState(frDetails *scaleiotypes.UploadComplianceTopol
 	state.Name = types.StringValue(frDetails.Name)
 	state.DefaultCatalog = types.BoolValue(frDetails.DefaultCatalog)
 	state.SourceLocation = types.StringValue(frDetails.SourceLocation)
+	if plan.Approve.IsNull() && frDetails.State == "needsApproval" {
```
Comment (rounak-adhikary) [inline powerflex/helper/firmware_repository_helper.go:36]: `approve` can never be null in the plan, it has a default value of `false` defined in the schema. Any reason why we are checking for null?

File: powerflex/provider/firmware_repository_resource.go [hunk 2]
```diff
@@ -152,6 +161,23 @@ func (r *firmwareRepositoryResource) Create(ctx context.Context, req resource.Cr
...
+			"Could not get the Firmware Repository Details",
+			err.Error(),
+		)
+		return
+	}
+
+	if frDetails.State == "copying" || frDetails.State == "pending" {
+		resp.Diagnostics.AddError(
+			"The Operation got timed Out",
+			err.Error(),
```
Comment (rounak-adhikary) [inline powerflex/provider/firmware_repository_resource.go:176]: This error is `nil`.

Comment (rounak-adhikary) [reply powerflex/provider/firmware_repository_resource.go:176]: Same in update function.

File: powerflex/helper/firmware_repository_helper.go (hunk 1 above)
Comment (AnikaAgiwal2711) [reply powerflex/helper/firmware_repository_helper.go:36]: when we import then that time this value is null

File: powerflex/provider/firmware_repository_resource.go (hunk 2 above)
Comment (AnikaAgiwal2711) [reply powerflex/provider/firmware_repository_resource.go:176]: Yeah. My bad

File: powerflex/helper/firmware_repository_helper.go (hunk 1 above)
Comment (rounak-adhikary) [reply powerflex/helper/firmware_repository_helper.go:36]: That makes sense. But do we even need the null check? We can always write the current data to state whether the resource is imported or not.

File: powerflex/provider/firmware_repository_resource.go (hunk 2 above)
Comment (AnikaAgiwal2711) [reply powerflex/provider/firmware_repository_resource.go]: Done

**Review by rounak-adhikary — APPROVED** [review]

**Review by shenda1 — APPROVED** [review]

**Comment by AnikaAgiwal2711** [comment]
run e2e test

**Comment by AnikaAgiwal2711** [comment]
https://osj-sio-03-prd.cec.lab.emc.com/job/Terraform-Pipelines/job/PowerFlex/job/tf-powerflex-pr/238/consoleFull

**Comment by shenda1** [comment]
run e2e test

**Pull request merged**